### ISSUE TITLE
cpp-linter workflows runs for every change

### ISSUE THREAD

**Issue by shenxianpeng** [cpp-linter/cpp-linter-action#205]
I guess your change here was to disable all cpp-linter runs only run when there is a change under `docs/examples/demo/`

https://github.com/cpp-linter/cpp-linter-action/blob/7ce9dc29bcbc91876bb1548b691fc96156981d8b/.github/workflows/cpp-linter.yml#L6

but right now, it seems cpp-linter runs and adds comments for everything. see https://github.com/cpp-linter/cpp-linter-action/commit/7ce9dc29bcbc91876bb1548b691fc96156981d8b#comments

**Comment by 2bndy5** [comment]
yes, that was the idea. But it didn't seem to work as expected. Maybe if we just remove that line, it would only run when there are changes to the demo sources.

[github docs for `on.<event>.path`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-paths)

Currently we are only using that workflow to show an example badge in the README. We don;t often modify the demo sources. So, how useful it is to even have both the CI workflow and the demo sources in this repo?

**Comment by 2bndy5** [comment]
We should also disable thread-comments for that workflow. That way we are less bothered by the posted comment.

**Comment by 2bndy5** [comment]
It might be getting triggered from master branch not main branch.

**Comment by 2bndy5** [comment]
Oh, I see. Actually, the comment comes from the [test CI workflow](https://github.com/cpp-linter/cpp-linter-action/blob/36abdeffc63476465e83e2a1e23680013d9fd26d/.github/workflows/run-test.yml) which calls the [test repo's reusable workflow](https://github.com/cpp-linter/test-cpp-linter-action/blob/master/.github/workflows/cpp-lint-action.yml).

So, maybe we don't run the test CI on push events?

**Comment by 2bndy5** [comment]
That change in 57cc38b62414a9cff629f90220851e18f80ee5ea might also fix the problem where the `latest` tag was pointing to the PR's base branch tip (instead of the PR's head branch tip).
